runtime/parachains/src/scheduler_parathreads/mod.rs

```diff
@@ -0,0 +1,170 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Polkadot is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Polkadot is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+use frame_support::pallet_prelude::*;
+use primitives::{CoreIndex, CoreOccupied, Id as ParaId, ParathreadClaim};
+use scale_info::TypeInfo;
+
+use crate::{
```

Comment:
As discussed, implementing `AssignmentProvider` for parathreads is already part of Anton's work.

Reply:
As I don't remember that being discussed at all, can you point me to where this decision was made? Not that I mind, I'd just like to know where I missed it.

Reply:
Obviously I remember that :joy: I meant before that.